### PR TITLE
Support passing Serializer / Deserializer instances when building pro…

### LINKED ISSUE #64: Support passing Serializer / Deserializer instances when building producer/consumer/stream
I would like to use my own deserializer.

Something like this:

```
import org.apache.kafka.common.serialization.Deserializer;
import java.util.Map;

public class MyDeserializer<T> implements Deserializer<T> {

  Class<T> classType;
  CustomDeserializer custDes;

  public MyDeserializer(Class<T> classType) {
    this.classType = classType;
    
   custDes = new CustDes.config();
   custDes.preregisterSomeClassesForPerformance(Class1.class,Class2.class);
  }

  @Override
  public void configure(Map<String, ?> map, boolean b) { }

  @Override
  public T deserialize(String s, byte[] bytes) {

    Object deserializedObject = custDes.deserialize(bytes);
    return classType.cast(deserializedObject);

  }

  @Override
  public void close() {}
}
```
 

Any advice on using the KafkaConsumer and directly deserialize the bytes from Kafka with my Deserializer.

I tried deserializing with BufferDeserializer first then with MyDeserializer wich works. But I would like to know if there's a way to do it without the intermediate BufferDeserializer. 

*Assuming data has been originally serialized with my custom serializer.

### PR DESCRIPTION
**Pull request by vietj**
…ducer/consumer/stream - fixes #64

### CODE REVIEW

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaProducerImpl.java [hunk 1]
```diff
@@ -57,10 +58,18 @@
     return createShared(vertx, name, () -> KafkaWriteStream.create(vertx, config, keyType, valueType));
   }
 
+  public static <K, V> KafkaProducer<K, V> createShared(Vertx vertx, String name, Properties config, Serializer<K> keyType, Serializer<V> valueType) {
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/client/producer/impl/KafkaProducerImpl.java]: keyType and valueType should be named keySerializer and valueSerializer

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaProducerImpl.java [hunk 2]
```diff
@@ -57,10 +58,18 @@
...
+    return createShared(vertx, name, () -> KafkaWriteStream.create(vertx, config, keyType, valueType));
+  }
+
   public static <K, V> KafkaProducer<K, V> createShared(Vertx vertx, String name, Map<String, String> config, Class<K> keyType, Class<V> valueType) {
     return createShared(vertx, name, () -> KafkaWriteStream.create(vertx, new HashMap<>(config), keyType, valueType));
   }
 
+  public static <K, V> KafkaProducer<K, V> createShared(Vertx vertx, String name, Map<String, String> config, Serializer<K> keyType, Serializer<V> valueType) {
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/client/producer/impl/KafkaProducerImpl.java]: Ditto as above

**Comment by vietj** [comment]
changes done

**Comment by vietj** [comment]
@ppatierno never mind, it was not really useful, providing a plain serializer or deserializer is simpler

**Comment by goshac** [comment]
Any chance for this PR to be merged?

**Comment by ppatierno** [comment]
Closing this one and using #136

**Pull request closed without merging**